src/handlers/pinnedMessages.ts

```diff
@@ -0,0 +1,128 @@
+    userForRoom: UserForRoom;
+};
+
+/**
+ * Type of Rocket.Chat messages
+ */
+export type RcMessage = {
+  _id: string
+  t?: string // Event type
+  rid: string // The unique id for the room
+  msg: string // The content of the message.
+  tmid?: string
+  ts: {
+    $date: string
+  }
+  mentions?: string[]
+  u: {
+    _id: string
+    username?: string
+    name?: string
+  }
+  // md?: any // The message's content in a markdown format.
+  pinned?: boolean
+  pinnedBy?: {
+    _id?: string
+    username?: string
+  }
+  drid?: string // The direct room id (if belongs to a direct room).
+  // attachments?: any[] // An array of attachment objects, available only when the message has at least one attachment.
+  reactions?: {
+    [key: string]: {
+      usernames: string[]
+    }
+  }
+}
+
+/**
+ * Reads the input file for messages, gets the mappings of each pinned message and returns this collection
+ * It also returns a dict to know which user needs to connect to pinned messages in a room
+ * @returns A collection with: PinnedMessages collection with messages IDs of rooms & Dict of rooms: users
+ */
+export async function getPinnedMessages(): Promise<PromiseResult> {
+  const pinnedMessages: PinnedMessages = {}
+  const userForRoom: UserForRoom = {}
+  const rl = new lineByLine(`./inputs/${entities.messages.filename}`)
+  let line: false | Buffer
+  while ((line = rl.next())) {
+    const message: RcMessage = JSON.parse(line.toString())
+    if (message.pinned) {
+      log.info(message._id)
+      const matrixRoomId = await getRoomId(message.rid)
+      const matrixMessageId = await getMessageId(message._id)
+      if (!matrixRoomId) {
+        log.warn(
+          `Room ${message.rid} has no mapping, thus no pinned messages.`
+        )
+        continue
+      }
+
+      if (message.pinnedBy !== undefined) {
+        log.info(matrixMessageId)
+        log.info(matrixRoomId)
+        log.info(message.pinnedBy._id)
+        if (message.pinnedBy._id !== undefined) {
+          userForRoom[matrixRoomId] = message.pinnedBy._id
+        }
+      }
+
+      if (!pinnedMessages[matrixRoomId]) {
+        pinnedMessages[matrixRoomId] = [];
+      }
+      if (matrixMessageId !== undefined ) {
+        pinnedMessages[matrixRoomId].push(matrixMessageId);
+      }
+    }
+  }
+  const promiseResult: PromiseResult = {pinnedMessages, userForRoom}
+  return promiseResult
+}
+
+/**
+ * Sets the m.room.pinned_events settings for rooms.
+ * @param pinnedMessages An object containing rooms and their pinned message, to be set in synapse
+ */
+export async function setPinnedMessages(
+  promiseResult: PromiseResult
+) {
+  const { userForRoom } = promiseResult;
+
+  for (const room in userForRoom) {
+    log.info("room")
+    log.info(room)
+
+    log.info("user")
+    log.info(userForRoom[room])
+    const userSessionOptions = formatUserSessionOptions(
+      (await getMapping(userForRoom[room],entities[Entity.Users].mappingType))?.accessToken || ''
+    )
```

Comment:
You could use `getUserSessionOptions(rcId: string)` instead:
```suggestion
    const userSessionOptions = getUserSessionOptions(userForRoom[room]) || ''
```